packages/paste-website/src/utils/RouteUtils.ts

```diff
@@ -18,6 +18,13 @@ export function getCurrentPathHash(): string {
   return window.location.hash;
 }
 
+export function getCurrentOrigin(): string {
+  if (!hasWindowObject()) {
+    return '';
+  }
+  return window.location.origin;
+}
+
```

Comment:
I'm now a little worried this might not work. If I use something like https://www.opengraph.xyz/, it doesn't get the image url because JavaScript needs to be downloaded and executed to update the meta tag after the DOM has loaded. So they grab the HTML and don't wait for JS to execute.

So any crawler that doesn't execute JS might not get our images.

But I do assume that most services do cater for the fact the this stuff might be reliant on client side rendering because React is a thing? 😕

Reply:
seems OK for FB https://developers.facebook.com/tools/debug/?q=https%3A%2F%2Fdeploy-preview-1661--paste-docs.netlify.app%2Fcomponents%2Fspinner

Reply:
Twitter claims it isn't working on this site https://cards-dev.twitter.com/validator, but I don't know.  We're using react-helmet to set the tag anyways, so we need JS regardless. Hmm

Reply:
According to the docs, it may just be because it's on a test deploy environment: https://developer.twitter.com/en/docs/twitter-for-websites/cards/guides/troubleshooting-cards

Reply:
I made a change that will make the URL return the correct value on SSR https://github.com/twilio-labs/paste/pull/1661/commits/8dc97fa75df0abc52048864a2dbf707195cef591